### DISCUSSION TITLE
emotion to css files during build

### DISCUSSION THREAD

**Discussion by yanai101** [emotion-js/emotion#3121] [Q&A]
Hi all - I have a question - is it possible to take emotion inline style and extract it to css files during the build process?

**Comment by wdcryer** [comment]
Based on [this answer](https://github.com/emotion-js/emotion/discussions/2827#discussioncomment-3446525) in another discussion, it seems unlikely that this will be supported by the library. 

>> Is there a plan? Is Emotion.js considering reintroducing some kind of [static extraction](https://github.com/emotion-js/emotion/blob/679ca35227f6806eb86f076946061c21ebc90bea/docs/extract-static.mdx) as recommended in the guide above?
> I don't think we plan to introduce such things. There are two primary reasons for this:
> 
> we don't have time to develop this
> it would likely make Emotion to be a different library
> Even though CSS-in-JS has its drawbacks it might be worth the tradeoff for many people. The lack of the required build step is a huge win for many projects.

At my company we made a prototype of a Babel plugin to do this, but it can only handle a subset of cases where the styles are not defined dynamically at runtime. The styles need to be statically extractable.
